File: flutter_app/lib/presentation/auth/role_selection_screen.dart
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';
import 'package:go_router/go_router.dart';
import 'auth_view_model.dart';
import 'login_screen.dart'; // For colors

class RoleSelectionScreen extends StatelessWidget {
  const RoleSelectionScreen({super.key});

  @override
  Widget build(BuildContext context) {
    return Consumer<AuthViewModel>(
      builder: (context, authViewModel, child) {
        final state = authViewModel.state;

        WidgetsBinding.instance.addPostFrameCallback((_) {
          if (state.isSuccess) {
            context.go('/home'); // Ensure you have this route defined
          }
        });

        return Scaffold(
          backgroundColor: bgColor,
          body: SafeArea(
            child: Center(
              child: Padding(
                padding: const EdgeInsets.all(24.0),
                child: Column(
                  mainAxisAlignment: MainAxisAlignment.center,
                  children: [
                    const Text(
                      "Complete Account",
                      style: TextStyle(
                        fontSize: 24,
                        fontWeight: FontWeight.bold,
                        color: darkBlue,
                      ),
                    ),
                    const SizedBox(height: 8),
                    const Text(
                      "Please select how you will be joining Golden Hand Caregivers.",
                      textAlign: TextAlign.center,
                      style: TextStyle(
                        fontSize: 16,
                        color: Colors.grey,
                      ),
                    ),
                    const SizedBox(height: 32),

                    if (state.errorMessage != null)
                      Padding(
                        padding: const EdgeInsets.only(bottom: 16.0),
                        child: Text(
                          state.errorMessage!,
                          style: const TextStyle(color: Colors.red, fontSize: 14),
                          textAlign: TextAlign.center,
                        ),
                      ),

                    if (state.isLoading)
                      const CircularProgressIndicator(color: logoGreen)
                    else ...[
                      _buildRoleButton(
                        context,
                        title: "I am a Client",
                        color: darkBlue,
                        onPressed: () => context.read<AuthViewModel>().saveGoogleUserRole("CLIENT"),
                      ),
                      const SizedBox(height: 16),
                      const Text("OR", style: TextStyle(color: Colors.grey, fontSize: 14)),
                      const SizedBox(height: 16),

                      _buildRoleButton(
                        context,
                        title: "I am a Caregiver",
                        color: logoGreen,
                        onPressed: () => context.read<AuthViewModel>().saveGoogleUserRole("CAREGIVER"),
                      ),
                      const SizedBox(height: 16),
                      const Text("OR", style: TextStyle(color: Colors.grey, fontSize: 14)),
                      const SizedBox(height: 16),

                      _buildRoleButton(
                        context,
                        title: "I am a Nurse",
                        color: const Color(0xFF8B5CF6), // Purple from Android
                        onPressed: () => context.read<AuthViewModel>().saveGoogleUserRole("NURSE"),
                      ),
                    ],
                  ],
                ),
              ),
            ),
          ),
        );
      },
    );
  }

  Widget _buildRoleButton(BuildContext context, {required String title, required Color color, required VoidCallback onPressed}) {
    return SizedBox(
      width: double.infinity,
      height: 60,
      child: ElevatedButton(
        onPressed: onPressed,
        style: ElevatedButton.styleFrom(
          backgroundColor: color,
          shape: RoundedRectangleBorder(
            borderRadius: BorderRadius.circular(12),
          ),
        ),
        child: Text(
          title,
          style: const TextStyle(
            fontSize: 16,
            fontWeight: FontWeight.bold,
            color: Colors.white,
          ),
        ),
      ),
    );
  }
}

File: app/src/main/java/com/kina/care/presentation/auth/RoleSelectionScreen.kt
package com.kina.care.presentation.auth

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.lifecycle.viewmodel.compose.viewModel

@Composable
fun RoleSelectionScreen(
    authViewModel: AuthViewModel = viewModel(),
    onRoleSelectedSuccess: () -> Unit
) {
    val authState by authViewModel.authState.collectAsState()

    LaunchedEffect(authState.isSuccess) {
        if (authState.isSuccess) {
            onRoleSelectedSuccess()
        }
    }

    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(Color(0xFFF8FAFC))
            .padding(24.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.Center
    ) {
        Text("ගිණුම සම්පූර්ණ කිරීම", fontSize = 24.sp, fontWeight = FontWeight.Bold, color = Color(0xFF0D3B66))
        Text(
            "ඔබ Golden Hand Caregivers හා සම්බන්ධ වන්නේ කුමන ආකාරයෙන්දැයි කරුණාකර තෝරන්න.",
            fontSize = 16.sp,
            color = Color.Gray,
            textAlign = TextAlign.Center,
            modifier = Modifier.padding(top = 8.dp, bottom = 32.dp)
        )

        if (authState.errorMessage != null) {
            Text(text = authState.errorMessage!!, color = Color.Red, fontSize = 14.sp, modifier = Modifier.padding(bottom = 16.dp))
        }

        if (authState.isLoading) {
            CircularProgressIndicator(color = Color(0xFF53A548))
        } else {
            Button(
                onClick = { authViewModel.saveGoogleUserRole("CLIENT") },
                modifier = Modifier.fillMaxWidth().height(60.dp),
                shape = RoundedCornerShape(12.dp),
                colors = ButtonDefaults.buttonColors(containerColor = Color(0xFF0D3B66))
            ) {
                Text("මම සේවාදායකයෙක් (Client)", fontSize = 16.sp, fontWeight = FontWeight.Bold)
            }

            Spacer(modifier = Modifier.height(16.dp))
            Text("OR", color = Color.Gray, fontSize = 14.sp)
            Spacer(modifier = Modifier.height(16.dp))

            Button(
                onClick = { authViewModel.saveGoogleUserRole("CAREGIVER") },
                modifier = Modifier.fillMaxWidth().height(60.dp),
                shape = RoundedCornerShape(12.dp),
                colors = ButtonDefaults.buttonColors(containerColor = Color(0xFF53A548))
            ) {
                Text("මම සාත්තු සේවකයෙක් (Caregiver)", fontSize = 16.sp, fontWeight = FontWeight.Bold)
            }

            Spacer(modifier = Modifier.height(16.dp))
            Text("OR", color = Color.Gray, fontSize = 14.sp)
            Spacer(modifier = Modifier.height(16.dp))

            Button(
                onClick = { authViewModel.saveGoogleUserRole("NURSE") },
                modifier = Modifier.fillMaxWidth().height(60.dp),
                shape = RoundedCornerShape(12.dp),
                colors = ButtonDefaults.buttonColors(containerColor = Color(0xFF8B5CF6))
            ) {
                Text("මම හෙදියක් (Nurse)", fontSize = 16.sp, fontWeight = FontWeight.Bold)
            }
        }
    }
}